rmw_zenoh_cpp/src/detail/liveliness_utils.cpp

```diff
@@ -441,7 +442,17 @@ Entity::Entity(
     // Append the delimiter unless it is the last component.
     this->liveliness_keyexpr_ += KEYEXPR_DELIMITER;
   }
-  this->guid_ = std::hash<std::string>{}(this->liveliness_keyexpr_);
+
+  // Here we hash the complete std::string that comprises the liveliness keyexpression
+  // into a GID that is associated with every entity in the system.  This is the GID that will be
+  // returned to the RMW layer as necessary.
+  simplified_XXH128_hash_t keyexpr_gid =
+    simplified_XXH3_128bits(this->liveliness_keyexpr_.c_str(), this->liveliness_keyexpr_.length());
+  memcpy(this->gid_, &keyexpr_gid.low64, sizeof(keyexpr_gid.low64));
+  memcpy(this->gid_ + sizeof(keyexpr_gid.low64), &keyexpr_gid.high64, sizeof(keyexpr_gid.high64));
```

Comment:
I think we should move this logic (ie, getting a `std::size_t` hash from a `std::string`) into `hash_gid()` defined in `liveliness_utils.hpp` since it will also be needed within `ServiceData::take_request`. See my comment below.

Reply:
This is now changed to use `hash_gid` in dbd7bfe141099c89ecf4f0b314144e5185dea3dc